App data plane session breaking change

yitaopan opened the pull request:
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

Commit:
Adding optional field to NetworkWatcherFlowLog

GuptaVertika committed:
Copy files from stable/2024-01-01

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

GuptaVertika committed:
Update version to stable/2024-03-01

Updated the API version from stable/2024-01-01 to stable/2024-03-01.

GuptaVertika committed:
Added tag for 2024-03-01 in readme file

Commit:
More verbose description of the field

Commit:
spell check

Commit:
Adding enabledFilteringCriteria field to 2024-03-01 version as well

Commit:
spell check

sindhualuguvelli1 committed:
Adding readonly field for list of attached AGCs

sindhualuguvelli1 committed:
fix

mansi-verma-1510 committed:
Revert "spell check"

This reverts commit 22d8ac44be4fef4bc17576928498dfaebbdb8caf.

mansi-verma-1510 committed:
Revert "Adding optional field to NetworkWatcherFlowLog"

This reverts commit e58d4457ca078b4953a27cbe4fef8b6977e91739.

mansi-verma-1510 committed:
Revert "More verbose description of the field"

This reverts commit bfbf2c869d6a012f1c727135a2f1ba30b7ec8830.

mansi-verma-1510 committed:
Revert "Adding enabledFilteringCriteria field to 2024-03-01 version as well"

This reverts commit a5b93be36120ad0d5b93b947c87a009414c01197.

mansi-verma-1510 committed:
Revert "spell check"

This reverts commit 0b79f8c0d773bc91dab62174fb4a5f2789321f5f.

mansi-verma-1510 committed:
Adding enabledFilteringCriteria field to 2024-03-01 NetworkWatcher Flowlogs (#29759)

* Adding enabledFilteringCriteria field to 2024-03-01 version as well

* Prettier NetworkWatcherFlowLogCreate file

* prettier files

* Prettier networkWatcher.json

---------

Co-authored-by: Mansi Verma <manverm@microsoft.com>

sindhualuguvelli1 committed:
Address comment

wdehrich committed:
Add Virtual Network Property PrivateEndpointsVNetPolicies (#29826)

* Add new properties

* Prettier

* Description must not match the name of the node it is supposed to describe. Node name:'destinationIPAddress' Description:'DestinationIPAddress Location: Microsoft.Network/stable/2024-03-01/privateLinkService.json#L813

* Update description of destination IP address

* privateEndpointVNetPolicies should not be read-only

* Revert change to PrivateLinkService

* Trailing whitespace

---------

Co-authored-by: Will Ehrich <wehrich@microsoft.com>

MaNehaKa committed:
Http DDoS ruleset and sensitivity swagger

MaNehaKa committed:
Fix using prettier

diagaval committed:
Add resiliency model support for Virtual Network Gateway (#29854)

* resiliency support:

* updated indentation

cshahra committed:
WAF AllowList / Exceptions (#29868)

erlicai-work committed:
Fix private endpoiont and private link services specs (#29830)

* fix pe and pls api specs

* fix model validation for service endpoint policy

* prettier

* prettier

wdehrich committed:
Add DestinationIPAddress to Private Link Service (#29833)

* Add new properties

* Prettier

* Description must not match the name of the node it is supposed to describe. Node name:'destinationIPAddress' Description:'DestinationIPAddress Location: Microsoft.Network/stable/2024-03-01/privateLinkService.json#L813

* Update description of destination IP address

* privateEndpointVNetPolicies should not be read-only

* Revert change to virtual network

* Fix

---------

Co-authored-by: Will Ehrich <wehrich@microsoft.com>

bewatersmsft committed:
Azure Firewall Custom Autoscale Configuration (#29933)

* Azure Firewall Manual Scale

* Provide autoscaleConfiguration int format and pretty

* Provide autoscaleConfiguration int format and pretty

* Prettier fixes

EugenioPena committed:
noHealthyBackensBehavior

Changed "NoHealthyBackendsBehavior" to "noHealthyBackendsBehavior" to fix some PowerShell issues.

appgwwaf committed:
fix tests api ver

EugenioPena committed:
Prettiercheck

Ran prettier check command.

bowenqin1 committed:
Fix example ApplicationSecurityGroupDelete.json found by Swagger Model Validation (#29988)

* Update ApplicationSecurityGroupDelete.json 2024-03-01

* prettier

* Fix previous version

bowenqin1 committed:
Mark Route property hasBgpOverride readonly

ajtms committed:
Fixes for azureFirewall spec examples for 2024-01-01 and 2024-03-01

ajtms committed:
Added missing long-running operation headers for 2024 azureFirewall spec examples

ajtms committed:
Ran prettier on some 2024-03-01 examples

ajtms committed:
Ran prettier on a few more 2024-03-01 examples

bowenqin1 committed:
prettier

ZhangYi-MS committed:
Update ipAllocation.json

ZhangYi-MS committed:
Update ipAllocation.json

ZhangYi-MS committed:
Update ipAllocation.json

ZhangYi-MS committed:
Update ipAllocation.json

ZhangYi-MS committed:
Update ipAllocation.json

ZhangYi-MS committed:
Update ipAllocation.json

ZhangYi-MS committed:
Update IpAllocationDelete.json

ZhangYi-MS committed:
Update IpAllocationDelete.json

ZhangYi-MS committed:
Update IpAllocationDelete.json

ZhangYi-MS committed:
Update IpAllocationDelete.json

ZhangYi-MS committed:
Update IpAllocationDelete.json

ZhangYi-MS committed:
Update IpAllocationDelete.json

ajtms committed:
Touching a file to trigger spec validation

ajtms committed:
Touching more files to trigger pipeline validation

ajtms committed:
Updated missing location header in 202 responses

ajtms committed:
Ran prettier

jbgorthy committed:
AVNM NSG + UDR Management Swagger changes for stable version (#30102)

* Avnm nsg and udr specs + examples.

* Run prettier check on network manager files.

* Add suppression for false alarm and update system data to use network.json ref.

* Add suppress keyword so lintdiff recognizes suppression.

---------

Co-authored-by: Jared Gorthy <jaredgorthy@microsoft.com>

Commit:
Fix examples

Commit:
change should be in RouteFilterRuleDelete.json, not RouteFilterDelete.json

Commit:
add RuleFilterDelete.json

Commit:
add 2024-03-01

Commit:
prettier

Commit:
took off too many lines

shengliumsft committed:
Add APIs to Get/Put/List shared key of vpn link connection for vwan VpnGateway (#30074)

* first commit

* run prettier

* fix checks

* fix checks

* resolve comments

* suppress error

Commit:
update description

amruthavvenugopal committed:
Update virtualNetworkGateway.json (#30096)

kamboj-prjwl committed:
2024-03-01 Fix Prettier Check

Commit:
0301

Commit:
0101

bowenqin1 committed:
Fix  virtualNetworkTapDelete example

bowenqin1 committed:
Fix example NetworkInterfaceTapConfigurationList & NetworkInterfaceTapConfigurationDelete

kamboj-prjwl committed:
Swagger Release 2024-03-01 - Fix Lint Diff

BlackRider97 commented on specification/app/Microsoft.App/routes.tsp:
```diff
@@ -21,7 +21,7 @@ alias SessionTraits = NoRepeatableRequests &
+interface SessionCodeExecution {
   #suppress "@azure-tools/typespec-azure-core/use-standard-operations" ""
   @doc("Execute code in a session.")
   @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/sessionPools/{sessionPoolName}/code/execute")
```
/codeExecute instead of `/code/execute` ?

BlackRider97 replied:
`/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/sessionPools/{sessionPoolName}` we can name this as session pool management endpoint as we are not doing anything special for it like returning these information back in response etc.

BlackRider97 commented on specification/app/Microsoft.App/routes.tsp:
```diff
@@ -34,34 +34,34 @@ interface CodeExecution {
   @doc("Get the content of the file.")
   @get
-  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/sessionPools/{sessionPoolName}/files/content/{filename}")
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/sessionPools/{sessionPoolName}/files/{filename}/content")
```
`/files/{fileName}/download` ? as it is more standard format and we anyway have equivalent upload file API

BlackRider97 commented on specification/app/Microsoft.App/models/code-execution.tsp:
```diff
@@ -36,25 +36,25 @@ union CodeInputType {
 
   @doc("Synchronous")
-  synchronous: "synchronous",
+  synchronous: "Synchronous",
```
we should also add type 'Asynchronous' as recently we added support for that. 

enableEgress & enableMultiArray fields are also missing.

BlackRider97 commented on specification/app/Microsoft.App/models/code-execution.tsp:
```diff
@@ -36,25 +36,25 @@ union CodeInputType {
 
 @doc("The properties of the code execution result.")
-model CodeExecutionResponseProperties {
+model SessionCodeExecutionResponseProperties {
```
We don't need to request param under properties as it can be top level properties too. Properties bag concept is applicable for ARM control-plane APIs not really applicable for data-plane APIs I mean have this flat like 

`{ "codeInputType": "Inline", "executionType": "Synchronous", "code": "string", "timeoutInSeconds": 0 } `

BlackRider97 replied:
SessionCodeExecutionResponse can simply extend SessionCodeExecutionRequest with bunch of ready-only properties in response. and `executionTimeInMilliseconds`  & other properties under `diagnosticInfo` property bag

BlackRider97 replied:
identifier should be ready-only in response which is populated from query param

BlackRider97 commented on specification/app/Microsoft.App/models/common.tsp:
```diff
@@ -69,7 +72,7 @@ model SessionPoolNameParameter {
 
 @doc("The code execution name path parameter.")
-model CodeExecutionFileNameParameter {
+model SessionResourceFileNameParameter {
```
fileName should be duplicated. 
<img width="398" alt="image" src="https://github.com/user-attachments/assets/6410126b-1117-4013-924d-1a7a8ad5e69b">

BlackRider97 commented on specification/app/Microsoft.App/models/code-execution-file.tsp:
```diff
@@ -5,33 +5,33 @@ using TypeSpec.Rest;
 @doc("Code execution file resource.")
 @parentResource(SessionPool)
 @resource("files")
-model CodeExecutionFile {
```
Currently we can upload/download/get file with relative file path also. how that is being handled under `fileName` parameter ?

BlackRider97 commented on specification/app/Microsoft.App/models/code-execution-file.tsp:
```diff
@@ -5,33 +5,33 @@ using TypeSpec.Rest;
+  sizeInBytes: int64;
 
   @doc("The date time when the file was last modified.")
-  lastModifiedTime: utcDateTime;
```
no createdAt time ?

mikeharder committed:
[tsp-client] Upgrade to 0.12.2 (#30417)

msyyc committed:
update tspconfig.yaml for python (#30408)

yitaopan committed:
Fix

jaskisin committed:
Changing Start/Stop Request body to optional (#30410)

Co-authored-by: Jaskirat Singh <jasksingh@microsoft.com>

priyankarking committed:
Priyankarking redisenterprise microsoft.cache 2024 09 01 preview (#30293)

* Copy files from preview/2024-06-01-preview

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to preview/2024-09-01-preview

Updated the API version from preview/2024-06-01-preview to preview/2024-09-01-preview.

* Added tag for 2024-09-01-preview in readme file

* Update swagger spec for new API version 2024-09-01-preview

* revert changes for updating the common type

* Update readme.python.md

* Revert usage of common types for operation result in order to avoid breaking change of format diff

* re-order request parameters

* Fix the type of checkNameAvailability

* Rename certain SKUs and remove resourcestate from accesspolicyassignment properties.

* make access policy name as ready only as it is fixed and can not be updated.

* make access policy name as ready only as it is fixed and can not be updated.

---------

Co-authored-by: Priyankar Gupta <priyagupta@microsoft.com>
Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>

mikeharder committed:
[dependency upgrade] prettier, typescript, vitest (#30421)

- Match microsoft/typespec repo

mikeharder committed:
[tsp-client] Automatically update daily via dependabot (#30420)

abhaygill338 committed:
Add List and Delete Subscription Changed Directory for new version 2024-08-01-preview. (#30257)

* Resolving S320 Item - Missing Get SubscriptionOperations for 2021-10-01 version.

* Fix pipeline suggestions.

* Minor fix.

* Make Get Subscription operations sync in this version.

* Added suppress rules for failing lint errors.

* Fix suppression Issues.

* Minor fix suppression GetOperation200.

* Address minor fixes.

* Address GetOperation200 changes.

* Update readme.md: attempt suppression fix by adding ["202"] to "responses" to path

* Add Initiate, Get and Accept Subscription Change Directory Api with new version.

* Cleaning up the changes in the add-ons inside swagger files.

* Add or modified readme file.

* Fix pipeline issues.

* Fixes on lint errors, model errors.

* Address PR changes.

* Address swagger PR fixes.

* Fixing swagger changes.

* Fix breaking changes paths to include into this new version.

* Fix Prettier Checks.

* Add or Update the parameter location in the swagger.

* Add suppressions for the lint errors + plus request schema changes.

* Suppress lint errors.

* Adding changes to revert initiate and accept subscription changed directory existing behavior swagger.

* Add new version, destination tenant and owner id for subscription initiate and accept change directory.

* Reorder new preview version changes with the latest GA version.

* Adding v5 type error response body.

* Add List and Delete Subscription Changed Directory for new version 2024-08-01.

* Fix prettier checks.

* Fix Lint Errors.

* Address Lint errors fix.

* Add Suppression, for Delete Response Codes and Delete Operation Responses.

* Address Fix in suppression.

* Update reasons on Suppresions - DeleteResponseCodes, DeleteOperationResponse.

---------

Co-authored-by: Abhay Pratap <abhaypratap@microsoft.com>
Co-authored-by: Konrad Jamrozik <kojamroz@microsoft.com>

mpothuganti committed:
Add new swagger for publicCloud.json for 2024-12-01 version (#30288)

* Add new swagger for publicCloud.json for 2024-12-01 version

* Fix readme file

* Update readme with latest changes from main

* Create sdk-suppressions.yaml

* update sdk-suppressions.yaml

* Update sdk-suppressions.yaml

* Update sdk-suppressions.yaml

* Update sdk-suppressions.yaml

* Update sdk-suppressions.yaml

---------

Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>
Co-authored-by: Alancere <804873052@qq.com>

ChenxiJiang333 committed:
update tspconfig.yaml for python (#30422)

* update tspconfig.yaml for python

* fix

* update tspconfig.yaml for python

* fix

* fix

yitaopan committed:
Fix api version

Geek12580 committed:
Support sslMode and useSystemTrustStore for MariaDB (#30186)

Co-authored-by: Shenwang Zeng <shenwangzeng@microsoft.com>

yitaopan committed:
Fix api version

yitaopan committed:
Fix examples

yitaopan committed:
Merge branch 'main' into app-data-plane-session-breaking-change

yitaopan committed:
Restore readme.md

yitaopan committed:
Merge branch 'app-data-plane-session-breaking-change' of https://github.com/yitaopan/azure-rest-api-specs into app-data-plane-session-breaking-change

yitaopan committed:
Fix examples

yitaopan committed:
Fix examples

yitaopan committed:
Fix examples

yitaopan committed:
Fix examples

yitaopan committed:
Fix examples

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix

ramneettung-msft committed:
Adding CreateCallFailed event to 2024-06-15-preview api-version. Related work item: 3867093 (#30435)

solankisamir committed:
ApiManagement : 2024-05-01 stable api-version release (#30321)

* Copy files from preview/2023-09-01-preview

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to stable/2024-05-01

Updated the API version from preview/2023-09-01-preview to stable/2024-05-01.

* Added tag for 2024-05-01 in readme file

* run prettier

fangru2023 committed:
set stable api version as global default. (#30311)

* set stable api version as global default.

* update.

catalinaperalta committed:
rename foundations error name (#30467)

Co-authored-by: Catalina Peralta <caperal@microsoft.com>

Commit:
read-only & required fix (#30450)

yuanzhuangyuanzhuang committed:
Add AOAI 2024-08-01 preview API release (#30430)

kamboj-prjwl committed:
Lint Diff fix

weidongxu-microsoft committed:
java, mgmt, configure tspconfig (#30470)

Han-msft committed:
Rename Face models and clients (#30354)

* Rename models

* Update python pkg version

* Reduce operation

Han-msft committed:
Rename CreateLivenessWithVerifySessionJsonContent (#30474)

* Rename CreateLivenessWithVerifySessionJsonContent

* Update tsv

yitaopan committed:
Use session pool management endpoint

Commit:
Introduce CosmosDB RP API 2024-09-01-preview Version (#29906)

* Copy files from preview/2024-05-15-preview

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to preview/2024-09-01-preview

Updated the API version from preview/2024-05-15-preview to preview/2024-09-01-preview.

* Added tag for 2024-09-01-preview in readme file

* run npx prettier

* run npx prettier

* run npx prettier

* add the v7 to the swagger (#29977)

* Update dataTransferService.json (#29994)

* CosmosDB: Add chaos fault API in cosmosDB RP (#29984)

* Changes for Chaos Fault RP API.

* Adding readme changes.

* fixing thePR validation error.

* Fixing spelling.

* Fixing the model.

* fixing the build issue.

* Fixing lintDiff.

* fixing lintdiff.

* changing the method in api.

* reveting the response,

* Reverting the rest api call method.

* updating the swagger changes,

* Swagger fix,

* more changes.

* Maxing suggested swagger changes.

* Fix Swagger validation changes.

* Fix lint diff.

* spelling mismatch.

* typo

* commit.

* Swagger fix.

* fixing name,

* Adding an all_of.

* adding api to cosmosdb.json

* Fixing the model.

* adding all changes to cosmos-db

* fix the response.

* removing the readme references.

* Adding a next

* fixing the response.

* adding all of.

* fixing the lint diff error.

* fxing the response

* removing value,

* fixing model.

* adding more items in list reponse.

* fixing api endpoint,

* endpoint update.

* Revert "endpoint update."

This reverts commit 868fef5a2caf8899451a84c43888c4551be5f360.

* reverting emaple files endpoint.

* fixing breaking chnages.

* types.json from v5-v1.

* Updating list example.

* moving all changes to chaosfault.jspn

* fxing the allof error.

* fix minor issue

* removing default.

* Revert "removing default."

This reverts commit 41149a22fcf0746eb0b8e454fc7ea8bd18aff930.

* Fixing lint diff.

* reverting changes.

* addig missing property

* Making supression rule.

* Adding another supression.

* Adding Next link property. (#30190)

* Removing supression and adding nextlink property.

* Add readme changes.

* fixing the json.

---------

Co-authored-by: Andrew Khoma <andrii.serg.khoma@gmail.com>
Co-authored-by: sarwesh <95228434+srwshkrshnn@users.noreply.github.com>
Co-authored-by: soagar <65682666+soagar@users.noreply.github.com>

Commit:
Introduce CosmosDB RP API 2024-08-15 Version (#29905)

* Copy files from stable/2024-05-15

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to stable/2024-08-15

Updated the API version from stable/2024-05-15 to stable/2024-08-15.

* Added tag for 2024-08-15 in readme file

* run npx prettier

* add suppression

* Removed preview from burst capacity description (#29941)

* Swagger changes for Restore with TTL Disabled (#29935)

* Swagger changes for Restore with TTL Disabled

* Format fix

* add the v7 to the swagger (#29976)

* fix online/offline example to adhere to swagger spec, fixing .net sdk build error

* nit run npx prettier

* Revert "add suppression"

This reverts commit 07e107227e7f64ac4d20802b304505fe70e19e30.

* update private endpoint connection to use standard ErrorResponse

* update private endpoint connections to use v5 types

* Revert "update private endpoint connections to use v5 types"
Causes powershell build to fail.
This reverts commit 49b4257ccf5b8b8443960bfc028f68b5ce523472.

---------

Co-authored-by: Richa Sharma <richasharma@microsoft.com>
Co-authored-by: minhokang242 <128573538+minhokang242@users.noreply.github.com>
Co-authored-by: Andrew Khoma <andrii.serg.khoma@gmail.com>

hahahahahaiyiwen committed:
AppConfiguration 2024-05-01 (#29825)

* Copy 2023-03-01

* Update 2024-05-01

* Address comments

davidemontanari committed:
[Microsoft.DeviceRegistry] Added spec for Microsoft.DeviceRegistry RP version 2024-09-01-preview (#30222)

* Added scaffolding for new API version

* Added definitions for API version 2024-09-01-preview

* Removed unused swagger file

* Fixed examples

* Fixed SpellCheck

* Added location for updates

* Added location for updates

* Added location for updates

* Fixed update definition

sssharma24 committed:
removing tags and location as they are alredy defined in CommonTrackedResource (#30327)

justinytchen committed:
justinytchen hybridcompute microsoft.hybrid compute 2024 07 31 preview (#29560)

Updated preview version with 2024-07-31-preview to add FirmwareProfile, HardwareProfile, and StorageProfile to Machine Properties in Hybrid Compute Machine Resource

---------

Co-authored-by: Justin Chen <juschen@microsoft.com>
Co-authored-by: Tim Lovell-Smith <tilovell@microsoft.com>

amruthashree18 committed:
Bug fix on 2022-05-13 API version to fix breaking change flag for PR 30290 (#30492)

* Bug fix on 2022-05-13 API version

* Bug fix on 2022-05-13 API version

* Bug fix on 2022-05-13 API version

mikeharder committed:
Remove file /npx added accidentally (#30494)

ArthurMa1978 committed:
Add client tsp for StandybyPool (#30488)

DapengLi2016 committed:
[VideoTranslation] Initial add TypedSpec and swagger (#29572)

mikeharder committed:
[package.json] Add engines section with minimum node/npm versions (#30520)

- Ensures devs working in repo use correct tool versions
- Also ensures dependabot uses these versions when updating package-lock.json

mikeharder committed:
Reduce minimum npm version from 10 to 9 (#30522)

- Dependabot still uses 9.6.5

mikeharder committed:
Update package-lock.json for engines change (#30524)

dependabot[bot] committed:
Bump @azure-tools/typespec-client-generator-cli from 0.12.2 to 0.13.0 (#30523)

Bumps [@azure-tools/typespec-client-generator-cli](https://github.com/Azure/azure-sdk-tools) from 0.12.2 to 0.13.0.
- [Release notes](https://github.com/Azure/azure-sdk-tools/releases)
- [Commits](https://github.com/Azure/azure-sdk-tools/commits)

---
updated-dependencies:
- dependency-name: "@azure-tools/typespec-client-generator-cli"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

stevemunk committed:
Removed the retired data service. (#30517)

Co-authored-by: Joel Hendrix <jhendrix@microsoft.com>

jlian committed:
Update IoT Operations examples (#30451)

* New service and SAT values in examples

* New configMap name in examples

* Update examples in the other folder too

qfai committed:
[developerhub] Update private preview 2024-05-01-preview (#30333)

* change typo

* fix ci issu

* fix model validation error

* add all quick start template type

---------

Co-authored-by: qfai <quke@microsoft.com>

FumingZhang committed:
Review request for Microsoft.ContainerService/aks to add version preview/2024-07-02-preview (#30395)

* Adds base for updating Microsoft.ContainerService/aks from version preview/2024-06-02-preview to version 2024-07-02-preview

* Updates readme

* Updates API version in new specs and examples

* Add driverType property to AKS AgentPoolGPUProfile (#30094)

* Add driverType property to AgentPoolGPUProfile

* Update managedClusters.json

* Update managedClusters.json

* remove outdated constrains about vm profile from description

* addressing breaking change concerns

* fix CI failures

* align with stable API

---------

Co-authored-by: Calvin S. <calvin197@ymail.com>
Co-authored-by: skuchipudi295 <128435195+skuchipudi295@users.noreply.github.com>

ChenxiJiang333 committed:
Update readme.python.md (#30528)

schaudhari6254888 committed:
[ServiceBus] 2024-01-01 (#30331)

* BASE commit

* Adding new api-version=2024-01-01

* Readme file updated

* Updates in failover api

* Readme correction'

* Readme correction again'

* Common-datatype updates

* Common-datatype removed ErrorResponse it is causing breaking change

* Updated common-types

* Updates

* more updates

---------

Co-authored-by: Your Name <you@example.com>

yitaopan committed:
Update namespace to Microsoft.App.DynamicSessions

yitaopan committed:
Merge branch 'main' into app-data-plane-session-breaking-change

rzhelenchuk committed:
Review request for Microsoft.Security to add version stable/2024-08-01 (#30235)

* Review request for Microsoft.Security to add version stable/2023-08-01

* fix

* fixes

* Change version

* fix perttier check

* update readme

* added custom recommendation type

* update reademe

* prettier

* prettier

* fixed modelling issue

* fix

* prettier

* fix

* remove next link

* revert

* fix

* addressed comments

* addressed comments

* addressed comments

* addressed comments

* fix model validation

* addressed comments

* Fixes after CR

* change to v1

* add body

* merge #/definitions/CustomRecommendationRequestBody to #/definitions/CustomRecommendation

* seperate request properties

* fix lint errors

* fixed systemdata examples

---------

Co-authored-by: Tal Cohen <talcohen@microsoft.com>
Co-authored-by: Elia Grady <eliagrady@users.noreply.github.com>
Co-authored-by: talco77 <158072483+talco77@users.noreply.github.com>
Co-authored-by: Elia Grady <elgrady@microsoft.com>

ChenxiJiang333 committed:
Update tspconfig for Python SDK (#30503)

* update tspconfig

* Update tspconfig.yaml

---------

Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>

yitaopan committed:
Add LRO accepted response for execute code

chrisradek committed:
Upgrade/typespec 0.60 (#30511)

Commit:
Fix lro wrong return type in AppConfig (#30548)

kazrael2119 committed:
update fabric tspconfig (#30489)

kazrael2119 committed:
add typespec-ts config into tspconfig for mongocluster (#30355)

* add typespec-ts config into tspconfig for mongocluster

* Update tspconfig.yaml

* Update client.tsp

* Update tspconfig.yaml

* Update tspconfig.yaml

---------

Co-authored-by: Qiaoqiao Zhang <55688292+qiaozha@users.noreply.github.com>

msyyc committed:
update tspconfig.yaml for python (#30551)

msyyc committed:
[mongocluster] Update tspconfig.yaml for SDK generation (#30553)

* [mongocluster] Update tspconfig.yaml

* Update tspconfig.yaml

---------

Co-authored-by: Weidong Xu <weidxu@microsoft.com>

kazrael2119 committed:
update standbypool tspconfig (#30550)

* add ts config to standbypool tspconfig

* Update tspconfig.yaml

---------

Co-authored-by: Qiaoqiao Zhang <55688292+qiaozha@users.noreply.github.com>

weidongxu-microsoft committed:
tspconfig, fix invalid service-dir (#30557)

XiaofeiCao committed:
fix (#30555)

vikaschoudharycs097 committed:
[Astronomer][New Version] Added optional renewalMode and endDate(readonly) fields in OfferDetails (#30334)

* [Astronomer][New Version] Added optional field autoRenew in LiftrBase.OfferDetails

* Added new version in readme file

* Fixing TypeSpec validation issue

* Fixing TypeSpec validation issue

* Fixed TypeSpec Validation issue

* [Fix]: "id" is required to return in response of GET/PUT resource calls

* [Fix]: "id" is required to return in response of GET/PUT resource calls

* Added "endDate" field in OfferDetails

[Updates]
----------------
- Added "endDate" field in OfferDetails
- Defined AutoRenewStatus enum for autoRenew field
- Defined versioning for stable version "2024-08-27"

* Updated AutoRenewStatus enum values type to "string"

* Updated the ordering of AutoRenewStatus enum values

[Update]
----------------
"Enable" is the default value of auto-renew, so placed at first
position.

* Replaced AutoRenewStatus enum by RenewalMode

bosesuneha committed:
Devhub api spec changes (#30007)

* added copy of last release

* new devhub changes
removed x-ms-client-flatten
added new routes for ADO OAuth
added azurePipelineProfile obj

ChenxiJiang333 committed:
[Container Orchestrator Runtime] Update tspconfig for Python SDK  (#30563)

* update tsp-config

* fix-format

ChenxiJiang333 committed:
[Computeschedule] Update tspconfig for Python SDK (#30576)

* update tspconfig

* update

* fix format

ravigautam1896 committed:
Release sql microsoft.sql 2024 05 01 preview (#30510)

* Adding folders and examples changes for Version 2024-05-01 preview version (#29200)

* Adds base for updating Microsoft.Sql from version preview/2023-08-01-preview to version 2024-05-01-preview

* Updates API version in new specs and examples

* Updates readme

* addressing comments

* resolving spell check errors

* resolving LroLocationheader errors

* supress errors from DatabaseAdvisiors.json

* reverting the supression

---------

Co-authored-by: Ravi Gautam <ravigautam@microsoft.com>

* Update for Import/Export with Managed Identity (#29596)

* Update for Import/Export with Managed Identity

* Fix the example reference and update description

* Set the additionalProperties type to string for DatabaseAdvisors API (#29823)

* Set the new additionalProperties type to string

* Remove the required type for additionalProperties, just keep them true

* Dev/ivarandj/instance pool operations api (#29932)

* Add instance pool operations swagger and examples

* Address prettier check errors

* Fix Swagger LintDiff errors

* Fix parameter names to match parameter patterns

* Remove validation for resourceGroupName parameter as it fails Swagger ModelValidation (and it should not be failing)

* Replace bool with enum (PR review remark)

---------

Co-authored-by: Ivan Arandjelovic <ivarandj@microsoft.com>

* Adding updated DatabaseOperations.json file (#29956)

* Adding updated DatabaseOperations.json file

* Trying to fix ModelValidation error

* Reverting testing change

---------

Co-authored-by: Roopesh Manda <rmanda@microsoft.com>

* Adding swagger files for minor changes related to release-sql-Microsoft.Sql-2024-05-01-preview (#30030)

* taking changes from dsmaindev side

* removing further DatabaseAdvisor.json changes

* resolving spellcheck validation

* resolving ModelValidation errors

* resolving avocado errors

* resolving LRO Headers Validation and xmsEnum Validation

* resolving indentation

---------

Co-authored-by: Ravi Gautam <ravigautam@microsoft.com>

* resolve merge conflict for cSpell.json (#30179)

Co-authored-by: Ravi Gautam <ravigautam@microsoft.com>

* reverting cspell.json changes (#30285)

* reverting cspell.json changes

* revert changes2

---------

Co-authored-by: Ravi Gautam <ravigautam@microsoft.com>

* Fix lintdiff enum errpr (#30383)

Co-authored-by: Ivan Arandjelovic <ivarandj@microsoft.com>

---------

Co-authored-by: Ravi Gautam <ravigautam@microsoft.com>
Co-authored-by: psong-ms <147777703+psong-ms@users.noreply.github.com>
Co-authored-by: UrosKrsticMIPerf <150959695+UrosKrsticMIPerf@users.noreply.github.com>
Co-authored-by: Ivan Arandjelovic <ivan.arandjelovic@gmail.com>
Co-authored-by: Ivan Arandjelovic <ivarandj@microsoft.com>
Co-authored-by: akromm <87152103+akromm@users.noreply.github.com>
Co-authored-by: Roopesh Manda <rmanda@microsoft.com>
Co-authored-by: Mike Harder <mharder@microsoft.com>

gjwoods committed:
Uploads - Update OperationId (#30495)

shalmolineogi committed:
Liftr-NewRelic Resubscribe API - 2024-10-01 stable version (#30425)

* adding new api stable version folder

* Updating swagger version in swagger and all example files

* adding new api to swagger and creating example, updating readme

* changing the version of common lib

* adding a new property offerId

* prettier-fix

* Create sdk-suppressions.yaml

* update sdk-suppressions.yaml

* Add refresh ingestion key API and its example

* Adding two new props in the response body of getbillinginfo api

---------

Co-authored-by: Shalmoli Neogi <sneogi@microsoft.com>
Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>
Co-authored-by: kazrael2119 <98569699+kazrael2119@users.noreply.github.com>
Co-authored-by: Dipesh Bhakat <107024030+dipeshbhakat-microsoft@users.noreply.github.com>
Co-authored-by: Apoorva Singh <apoorvasingh@microsoft.com>

kamboj-prjwl committed:
Updated Codeowners Network (#30581)

schaudhari6254888 committed:
[Relay] 2024-01-01 (#30426)

* base commit

* New api 2024-01-01 version added

* Example file updated

* readme updates

* Create sdk-suppressions.yaml

* Update sdk-suppressions.yaml

---------

Co-authored-by: Your Name <you@example.com>
Co-authored-by: kazrael2119 <98569699+kazrael2119@users.noreply.github.com>
Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>

ms-mohammedali committed:
[LCM] Update event grid schema to include cold tier stats (#30097)

* [LCM] Update event grid schema to include cold tier stats

* Generated swagger

* Updated storage.json file

AndrewNivenMSFT committed:
Add Azure Programmable Connectivity 2024-01-15-preview management API (#30537)

* Add Azure Programmable Connecittivity 2024-01-15-preview management API

* Remove deprecated examples-directory config option

jaskisin committed:
Adding Operation Id as required by PS (#30586)

Co-authored-by: Jaskirat Singh <jasksingh@microsoft.com>

priyavj08 committed:
added a new stable api version 2024-04-15 (#30439)

* publishing 2024-04-15 api

* added input file details for 2024-04-15

* modified readme for SDK generation error

* removed extra ```

* committing the base version which is 2023-09-01

* Revert "committing the base version which is 2023-09-01"

This reverts commit 421e26083e6ba81df1506cb73260a3b6605ef5ce.

---------

Co-authored-by: Priya Vijayaraghavan <priyavij@microsoft.com>

snehanagendra committed:
Add a new 2024-09-01-preview dataplane API version to the DevCenter (#30359)

* add 2024-09-01-preview to main.tsp

* Add force option for environment deletion, add apis for get/list snapshots and restore to snapshot

* update readme

* commit changes  by npx tsv specification/devcenter/DevCenter

* address review feedback (add format info on timestamps, rename to restoreSnapshot, update operation result)

* remove stale example file

* Add parameters file

* change alias to model per PR feedback

* update version in client.tsp

---------

Co-authored-by: Sneha Nagendra <snnagen@microsoft.com>

yuc-Li committed:
Update ClusterPatchParameters desc (#30453)

[HDInsight] Update ClusterPatchParameters description

gigigoo0 committed:
Microsoft.maps 2024 07 01 preview (#30251)

* Adds base for updating Microsoft.Maps from version stable/2023-06-01 to version 2024-07-01-preview

* Add account locations

* update common definition refs

* apply checks

* update account names pattern

* fix pattern for openapi

* Update sdk-suppressions.yaml

---------

Co-authored-by: Gigi Grajo <gigigrajo@microsoft.com>
Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>

bjqian committed:
Add api-version 2024-08-01-preview for Azure SignalR and Azure Web PubSub (#30274)

* Init 2024-08-01-preview

* Update to 2024-08-01-preview

* Add ApiTest

* remove jsonencode func and update tests for privateEndpointConnection and certificates

* auto approve shared private link resource

* Fix spelling

* Add missing properties in ApiTest

* Update and fmt tf for wps

* keep groupId required

---------

Co-authored-by: Heng Lu <henglu@microsoft.com>

DapengLi2016 committed:
[VideoTranslation]Rename operation ID for translation/iteration creation API due to we don't support replace in the API. (#30575)

* Rename operation ID for translation/iteration creation API due to we
don't support replace in API.

* Supress operation ID change warning.

* Supress typespec validation warning by PR validation.

yitaopan committed:
Merge branch 'main' into app-data-plane-session-breaking-change

mikeharder committed:
[protected-files] If no changes, write to log instead of notice (#30617)

- Notice is too noisy for non-problem logs

weidongxu-microsoft committed:
java, webpubsub, readme config for resource-collection-associations (#30620)

yitaopan committed:
Review

yitaopan committed:
Review

yitaopan committed:
Review

cataggar committed:
reorg tsp files for Microsoft.AVS.Management (#30387)

* create files for reorg

* add boilerplate usings

* add back routes.tsp

* split up routes

* tsp format

* moved 300 lines of models

* split up models

* remove unreferenced ClusterResourceId

* Update sdk-suppressions.yaml

* Update sdk-suppressions.yaml

---------

Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>
Co-authored-by: kazrael2119 <98569699+kazrael2119@users.noreply.github.com>

JimSuplizio committed:
Update CodeownersLinter for net6 to net8 update (#30612)

jpalvarezl committed:
[OpenAI] Added missing object field in `EmbeddingItem` (#30446)

* Added missing object field in Embedding object

* Using single quotes to prevent having to escape

* Model validation check fix

* Corrected JSON

cdxu-147 committed:
change segment in security policies (#30602)

* change segment in security policies

* set provisioing state as readonly

* change segment in security policies

* set provisioing state as readonly

* undo api change

* undo api change

* add use-read-only-status-schema to resolve provisioningStateMustBeReadOnly Violation

* add newly generated swagger

gjwoods committed:
AOAI Uploads Documentation - Improve Documentation  (#30629)

andborja committed:
[Microsoft.Monitor] PipelineGroups v2024-10-01-preview (#30595)

* [Microsoft.Monitor] PipelineGroups v2024-10-01-preview

Moving from RPSaaSMaster, this new version implements the new folder structure for independent services.

Work item:
https://msazure.visualstudio.com/One/_workitems/edit/29413029

kazrael2119 committed:
update machinelearning readme.typescript,md (#30227)

Co-authored-by: Qiaoqiao Zhang <55688292+qiaozha@users.noreply.github.com>
Co-authored-by: ZhidaLiu <zhili@microsoft.com>

xiaoxuqi-ms committed:
[Reviewed in Private repo] Add new api version: 2024-06-01-preview for two new features (#29991)

* copy previous version as first commit

* update version and readme

* add new feature: optional database number and canary maintenance

* fix prettier check

* update readme

* fix model validation

* fix model validation

* fix prettier check

* add ARM Strong test

yitaopan committed:
Update route

yitaopan committed:
Upload file

yitaopan committed:
Download file

nehalR12 committed:
making changes in documentation according to GA. (#30636)

Co-authored-by: Nehal Reshu <nehalreshu@microsoft.com>

ryazhang-microsoft committed:
ACL Deployment Customizations writeLBAddressPrefix (#30480)

* add field in request body writeLBAddressPrefix

* add workerThreads

* add example

* Update ConfidentialLedger_Create.json

---------

Co-authored-by: ryazhang <ryazhang.microsoft.com>

stevemunk committed:
Added retirement banners. (#30597)

dazilb committed:
Add codeowners for Video Indexer (#30534)

Adding Omer, Tal, Raz, Einav and Daria to VI Code owners

gerrytan committed:
New RP Microsoft.AzureTerraform.Management - export Azure resources to Terraform HCL (#30455)

* New RP: Microsoft.AzureTerraform (ver=2023-07-01-preview)

* prettier

* remove x-ms-discriminator-value

* readme.md: add `openapi-subtype`

* Fix LintDiff failure

* Result configuration type change from object to string

* Remove properties: `subscriptionId` and `environment`

* Change `errors` from list of string to list of `ErrorResponse`

* Add operations endpoint

* Update

* Update

* AzureTerraform RP: Sync the MS version with the private version

* Terraform: Change `exportEerraform` endpoint to LRO (#18782)

* terraform: Modify the `exportTerraform` RT to be LRO

* Update

* prettier

* spell

* CI validation

* More CI issue fixes

* typo

* Add readme.go.md

* typo

* prettier

* suppression

* Update suppression

* AzureTerrform: Remove the parallelism property (#18909)

* exportTerraform: Change resourceId -> resourceIds; Add `maskSensitive`; Change `fullConfig` defaults (#19178)

* exportTerraform: Change resourceId -> resourceIds; Add `maskSensitive`; Change `fullConfig` defaults

* rename

* rename

* rename

* Specify LRO model (#19401)

* Add up the `resourceId` in `OperationStatus` model (#19450)

* Add `uri` format (#19455)

* Add format url

* update

* TypeSpec for Microsoft.AzureTerraform (#19390)

* TypeSpec for Microsoft.AzureTerraform

* Sorted swagger file, and added typespec generated with tsp-client

* revert export.json in prep of merging

* Sorted export.json for easy diff on future typespec -> swagger

* Rerun swagger -> typespec conversion after merging latest RPSaaSMaster,
updated namespace

* tsp compile for TypeSpec -> swagger

* Refactored typescript to minimise swagger diff

* Spread ErrorResponse into OperationStatus model

* Fixed enums, use doc decorator

* Update exportTerraform path desc

* Update doc for ExportResource model

* LintDiff fix: remove unnecessary auth

* Add resourceId prop to OperationStatus model, fixed char casing so they're consistent throughout

* Made resourceId read only

* Check `Swagger BreakingChange`: minimise diff on x-ms-enum

* TypeSpec Validation fix: rename dir and add missing @doc

* Upgrade/rp tsp 0.60 (#19551) -- applying only for specification/terraform

* Revert accidental changes to cSpell.json while merging

* Added tf files used for armstrong testing

---------

Co-authored-by: magodo <wztdyl@sina.com>

mikeharder committed:
[TypeSpecRequirement] Format document in tests (#30668)

zman-ms committed:
[AKS][Fleet] 2024-05-02-preview API (#28971)

- 2024-05-02-preview api
- introduce Kubernetes Fleet Manager AutoUpgradeProfile resource

---------

Co-authored-by: Stéphane Erbrech <stephane.erbrech@gmail.com>
Co-authored-by: Minh Nguyen <comrc019@icloud.com>
Co-authored-by: nguyenmin <nguyenmin@microsoft.com>
Co-authored-by: daru__ <ptd2108@columbia.edu>

avjerafi committed:
Add new API version 2024-01-01-preview to Log Search Alerts (Scheduled Query Rules) (#30312)

* Copy files from stable/2023-12-01

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to preview/2024-01-01-preview

Updated the API version from stable/2023-12-01 to preview/2024-01-01-preview.

* Added tag for 2024-01-01-preview in readme file

* Add EventLogAlert kind

* Add Dynamic Threshold related fields

* Added examples

* Fix existing typo: correcting field name to "resolveConfiguration"

* Prettier fixes

* Added pattern to ruleName field

* Fixed rule name pattern

* Update sdk-suppressions.yaml

---------

Co-authored-by: Avi Jerafi <avjerafi@microsoft.com>
Co-authored-by: Yuchao Yan <yuchaoyan@microsoft.com>

Juliehzl committed:
Make the header optional for source control (#30233)

kanikagupta11 committed:
Swagger changes to support pageSize in ServiceNowV2 (#30583)

* Update Pipeline.json

* Update Pipeline.json

FumingZhang committed:
Review request for Microsoft.ContainerService/aks to add version stable/2024-08-01 (#30657)

* Adds base for updating Microsoft.ContainerService/aks from version stable/2024-07-01 to version 2024-08-01

* Updates readme

* Updates API version in new specs and examples

* add ignore-pod-disruption-budget option for agentpool delete (#30614)

Co-authored-by: Tong Chen <tonche@microsoft.com>

---------

Co-authored-by: Tong Chen <99183466+tonychen15@users.noreply.github.com>
Co-authored-by: Tong Chen <tonche@microsoft.com>

abrinegarMicrosoft committed:
Abrinegar/azuredatatransfer/adding uri properties (#30516)

* Copy files from stable/2024-05-07

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to stable/2024-09-11

Updated the API version from stable/2024-05-07 to stable/2024-09-11.

* Added tag for 2024-09-11 in readme file

* Adding Uri properties

* Changed uri from strings to urls

* added url format, changed type back to string

tianxin-ms committed:
Cognitive Services 2024-10-01 (#30491)

* Copy files from preview/2024-06-01-preview
Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to stable/2024-10-01
Updated the API version from preview/2024-06-01-preview to stable/2024-10-01.

* Added tag for 2024-10-01 in readme file

* Cognitive Services 2024-10-01

yitaopan committed:
Merge branch 'main' into app-data-plane-session-breaking-change

yitaopan committed:
Fix TypeSpec validation

yitaopan committed:
Fix SpellCheck

yitaopan committed:
Fix SpellCheck

yitaopan committed:
Fix LintDiff

yitaopan committed:
Fix examples

yitaopan committed:
Fix examples

yitaopan committed:
Fix examples

yitaopan committed:
Fix LinkDiff

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix LRO

yitaopan committed:
Add example

yitaopan committed:
Fix examples

yitaopan committed:
Fix examples

yitaopan committed:
Remove NetworkStatus

yitaopan committed:
Update doc

yitaopan committed:
Update path

yitaopan committed:
Update code execution resposne

yitaopan committed:
Update path

yitaopan committed:
Change api version from 2024-09-01 to 2024-10-02-preview

yitaopan committed:
Fix readme

yitaopan committed:
Fix example

yitaopan committed:
Fix LRO status

yitaopan committed:
identifier to sessionId

yitaopan committed:
Add error response to code execution

yitaopan committed:
Update SessionResourceFile:

yitaopan committed:
Update code execute

yitaopan committed:
Update code execute

yitaopan committed:
Use Foundations.OperationState

yitaopan committed:
Update code execute

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Use query param for file path

yitaopan committed:
Use query param for file path

yitaopan committed:
Fix tests

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix doc

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Add content type

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix example

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix

yitaopan committed:
Fix examples

yitaopan committed:
Fix naming

yitaopan committed:
Model code execution as resource

yitaopan committed:
Record unknown type

yitaopan committed:
Use standard operation in GetFileContent action

yitaopan committed:
Fix
